cli/cmd/version.go

```diff
@@ -67,8 +88,123 @@ func IsHappyOutdated(cmd *cobra.Command) (bool, *util.Release, *util.Release, er
 	cliVersion := util.GetVersion()
 	latestAvailableVersion, err := GetLatestAvailableVersion(cmd)
 	if err != nil {
-		return false, cliVersion, latestAvailableVersion, err
+		log.Debug("Error getting latest available version. Will fail silently.")
+		//return false, cliVersion, latestAvailableVersion, err
+		return false, cliVersion, cliVersion, nil // Lie.
 	}
 
 	return !cliVersion.Equal(latestAvailableVersion), cliVersion, latestAvailableVersion, nil
 }
+
+func WarnIfHappyOutdated(cmd *cobra.Command) {
+
+	outdated, cliVersion, latestAvailableVersion, err := IsHappyOutdated(cmd)
+
+	if err != nil {
+		log.Errorf("Error checking for latest available version number: %v", err)
+		return
+	}
+
+	if outdated {
+		log.Warnf("This copy of Happy CLI is not the latest available. CLI version: %s  Latest available version: %s\n", cliVersion.Version, latestAvailableVersion.Version)
+		log.Warn("To update on Mac, run:  brew upgrade happy")
+	}
+
+}
+
+func CreateHappyVersionLockfileHandler(cmd *cobra.Command) error {
+	happyConfig, err := config.GetHappyConfigForCmd(cmd)
+	if err != nil {
+		return err
+	}
+
+	currentVersion := util.GetVersion()
+	projectRoot := happyConfig.GetProjectRoot()
+
+	requestedVersion, _ := cmd.Flags().GetString("version")
+
+	if requestedVersion == "" {
+		requestedVersion = currentVersion.Version
+	}
+
+	path, version, err := createHappyVersionLockFile(projectRoot, requestedVersion)
+
+	if err != nil {
+		return err
+	}
+
+	fmt.Fprintf(cmd.OutOrStdout(), "Created %s locking Happy to version %s\n", path, version)
+
+	return nil
+
+}
+
+func createHappyVersionLockFile(projectRoot string, requestedVersion string) (string, string, error) {
+
+	versionFile, err := config.NewHappyVersionLockFile(projectRoot, requestedVersion)
+	if err != nil {
+		return "", "", err
+	}
+
+	err = versionFile.Save()
+	if err != nil {
+		return "", "", err
+	}
+
+	return versionFile.Path, versionFile.HappyVersion, nil
+}
+
+func VerifyHappyIsLockedVersion(cmd *cobra.Command) (bool, string, string, error) {
```

Comment:
```suggestion
func VerifyHappyIsLockedVersion(cmd *cobra.Command) (string, string, error) {
```

I think having an error or not an error is good enough to verify. I think you can safely remove this bool

Reply:
In this case, the error would represent an error checking (likely reading the lock file). The bool says "Yes I checked, and yes it is the locked value (or it's not)".